### PR TITLE
Fuzz v15

### PR DESCRIPTION
**Pull request by catenacyber**
Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/2859

Describe changes:
- Adds six fuzz targets
- Change compilation scheme so as to compile fuzz targets (option --enable-fuzztargets for configure)

Modifies #4060 with 
- adding more `const` keywords for appveyor (different ifdef)

Some commits can be cherry-picked independently : log, config, style

### CODE REVIEW

File: src/suricata.h [hunk 1]
```diff
@@ -135,7 +135,7 @@ typedef struct SCInstance_ {
     enum RunModes aux_run_mode;
 
     char pcap_dev[128];
-    char *sig_file;
+    const char *sig_file;
```
Comment (victorjulien) [inline src/suricata.h]: this is not related to app-layer, so should go into its own commit

File: src/Makefile.am [hunk 2]
```diff
@@ -519,13 +525,88 @@ EXTRA_DIST = tests
...
 AM_CPPFLAGS = $(all_includes)
 
+suricata_SOURCES = suricata.c suricata.h $(COMMON_SOURCES)
+
 # the library search path.
 suricata_LDFLAGS = $(all_libraries) ${SECLDFLAGS}
-suricata_LDADD = $(HTP_LDADD) $(RUST_LDADD)
+suricata_LDADD = $(RUST_SURICATA_LIB) $(HTP_LDADD) $(RUST_LDADD)
+
+fuzz_applayerprotodetectgetproto_SOURCES = tests/fuzz/fuzz_applayerprotodetectgetproto.c $(COMMON_SOURCES)
```
Comment (victorjulien) [inline src/Makefile.am]: would it be possible to move those into a src/fuzz/Makefile.am or fuzz/Makefile.am? or perhaps a fuzz/applayerprotodetectgetproto/Makefile.am?

File: src/suricata.h (hunk 1 above)
Comment (catenacyber) [reply src/suricata.h]: Actually, this cannot be made `const` as it will be used in some cases in argument for `dirname`

File: src/Makefile.am (hunk 2 above)
Comment (catenacyber) [reply src/Makefile.am]: I would love to.
The only way I see to achieve that is to compile `$(COMMON_SOURCES)` ie every source file except suricata which has `main` into a library (a convenience one).
I spent some hours trying and this is the best workaround I found, though it is not pretty.
The main problem is that C and Rust files depend on one another...

Any help from an autotool expert is welcome here :-)
